test/test_jit.py

```diff
@@ -6261,6 +6261,17 @@ def fn():
         with self.assertRaisesRegex(torch.jit.Error, "must not be zero"):
             fn()
 
+    def test_script_for_in_list(self):
```

Comment:
i think this is probably covered by `get_sum_list_fn` tests ( https://github.com/pytorch/pytorch/blob/master/test/test_jit.py#L3653 ) , but I might be mistaken.